modules/intensity_transform/test/test_intensity_transform.cpp

```diff
@@ -198,4 +198,56 @@ TEST(intensity_transform_contrastStretching, accuracy)
     EXPECT_LE(cvtest::norm(res, expectedRes, NORM_INF), 1);
 }
 
+#ifdef HAVE_EIGEN
+namespace
+{
+double RMSE(const Mat_<Vec3b>& ref, const Mat_<Vec3b>& cur)
+{
+    double rmse = 0;
+
+    for (int i = 0; i < ref.rows; i++)
+    {
+        for (int j = 0; j < ref.cols; j++)
+        {
+            for (int k = 0; k < ref.channels(); k++)
+            {
+                rmse += (ref(i,j)(k) - cur(i,j)(k)) * (ref(i,j)(k) - cur(i,j)(k));
+            }
+        }
+    }
+
+    return std::sqrt(rmse / (ref.rows*ref.cols*ref.channels()));
+}
+}
+#endif
+
+TEST(intensity_transform_BIMEF, accuracy)
+{
+#ifdef HAVE_EIGEN
+    const std::string directory = "intensity_transform/BIMEF/";
+    const std::vector<std::string> testFilenames = {
+        "P1000205_resize", "P1010676_resize", "P1010815_resize"
+    };
+
+    for (const auto& filename : testFilenames)
```

Comment:
It would be nice to use test parameter instead:

```
typedef testing::TestWithParam< std::string > intensity_transform_BIMEF;

TEST_P(intensity_transform_BIMEF, accuracy)
{
    std::string filename = GetParam();
    ...
}

const string BIMEF_accuracy_cases[] = {
    "P1000205_resize",
    "P1010676_resize",
    "P1010815_resize"
};

INSTANTIATE_TEST_CASE_P(/*nothing*/, intensity_transform_BIMEF,
        testing::ValuesIn(BIMEF_accuracy_cases)
);
```

Reply:
Done.